### PR TITLE
Story at1 184

### PR DESCRIPTION
**Pull request by apurvapatkar**
- Added ReleaseResources functionality. 
- Written the related test cases.

### CODE REVIEW

File: tmcprototype/CentralNode/CentralNode/CONST.py [hunk 1]
```diff
@@ -41,6 +42,10 @@
 STR_FALSE = "False"
 STR_DISH_DUPLICATE = "List of the dishes that are already allocated: "
 STR_ASSIGN_RESOURCES_SUCCESS = "Resources are assigned successfully"
+STR_FALSE_TAG = "Resources cannot be released since ReleaseALL Tag is False"
```
Comment (adityadangeska) [inline tmcprototype/CentralNode/CentralNode/CONST.py]: Consider rewording the message to "False in ReleaseALL tag is not yet supported."

File: tmcprototype/CentralNode/CentralNode/CentralNode.py [hunk 2]
```diff
@@ -511,6 +511,96 @@ def AssignResources(self, argin):
...
+        except ValueError:
+            self.dev_logging(CONST.ERR_INVALID_JSON, int(tango.LogLevel.LOG_ERROR))
+            self._read_activity_message = CONST.ERR_INVALID_JSON
+        except KeyError:
+            self.dev_logging(CONST.ERR_JSON_KEY_NOT_FOUND, int(tango.LogLevel.LOG_ERROR))
+            self._read_activity_message = CONST.ERR_JSON_KEY_NOT_FOUND
+
+        argout = {
+            "ReleaseAll" : True,
+            "receptorIDList" : []
```
Comment (adityadangeska) [inline tmcprototype/CentralNode/CentralNode/CentralNode.py]: If the API fails, the ReleaseALL tag should be False and Receptor id should have list of resources which are not released.

File: tmcprototype/CentralNode/CentralNode/CONST.py (hunk 1 above)
Comment (apurvapatkar) [reply tmcprototype/CentralNode/CentralNode/CONST.py]: Done.

File: tmcprototype/CentralNode/CentralNode/CentralNode.py (hunk 2 above)
Comment (apurvapatkar) [reply tmcprototype/CentralNode/CentralNode/CentralNode.py]: Yes. We have changed the functionality as per suggestion.

File: tmcprototype/CentralNode/CentralNode/CONST.py (hunk 1 above)
Comment (apurvapatkar) [reply tmcprototype/CentralNode/CentralNode/CONST.py]: Th commit id: [485e1cec5f7608502b3bafde2c932940c3532cfb](https://github.com/ska-telescope/tmc-prototype/pull/21/commits/485e1cec5f7608502b3bafde2c932940c3532cfb)

File: tmcprototype/CentralNode/CentralNode/CentralNode.py (hunk 2 above)
Comment (apurvapatkar) [reply tmcprototype/CentralNode/CentralNode/CentralNode.py]: The commit id: [485e1cec5f7608502b3bafde2c932940c3532cfb](https://github.com/ska-telescope/tmc-prototype/pull/21/commits/485e1cec5f7608502b3bafde2c932940c3532cfb)

**Pull request merged**